Fix non-reloc install & fix install into different prefixes for MoarVM, NQP, Rakudo

patrickbkr opened the pull request:
This PR is part of a set of changes in Rakudo, NQP and this repository. Only merge as a whole.

PatZim committed:
Add macro to escape C string literals

This is needed in the rakudo C build dir perl6-m runner.

PatZim committed:
Fix relocatability configuration

The logic didn't check the correct flag and thus always ended up
configuring the relocatable version.

PatZim committed:
Make non-relocatability also work with relative --with-nqp

vrurg commented on lib/NQP/Config.pm:
```diff
@@ -426,10 +426,11 @@ sub configure_relocatability {
 
     # Relocatability is not supported on AIX.
-    $self->{no_relocatable} ||= !!( $^O =~ /^(?:aix|openbsd)$/ );
+    $self->{options}->{"no-relocatable"} ||= !!( $^O =~ /^(?:aix|openbsd)$/ );
```
Please, don't change `options` values. I only allowed one unfortunate case of setting `gen-nqp` in Configure.pl. But otherwise they must remain unchanged as the ground we use to build up all other constructs.

patrickbkr replied:
Fixed that.

PatZim committed:
Move rakudo specific bits of reloc handling to rakudo

PatZim committed:
Default to doing a non-relocatable build

Non relocatability is the more common case with less caveats.
Remove the --no-relocatable config, add --relocatable instead.

vrurg reviewed (approved):
Looks fine. You should be able of merging this on your own, unless I'm mistaken. It's better be done synchronously with `nqp` and `rakudo`.

PatZim committed:
Fix a typo

PatZim committed:
Fix is_executable on Windows

Sometimes paths that include the extension are passed. In those cases we
must also try the path name without adding extensions.

PatZim committed:
Merge branch 'c-escaper' into reloc-fix

PatZim committed:
Add a C escaper to Config

So it's usable from the perl side as well.

PatZim committed:
Don't change option values

vrurg reviewed (approved):
Looks great. Thanks!

PatZim committed:
Use speaking values for the reloc flag

PatZim committed:
Fail when being requested to do a relocatable build and we can't

That's the case on AIX and OpenBSD at the moment.

vrurg commented on lib/NQP/Config.pm:
```diff
@@ -423,13 +423,13 @@ sub configure_jars {
-        $self->{config}->{relocatable} = $self->{options}->{relocatable} ? 'reloc' : 'nonreloc';
+    # Relocatability is not supported on AIX and OpenBSD.
+    if ( $^O =~ /^(?:aix|openbsd)$/ && $self->{options}->{relocatable} ) {
+        die 'Relocatability is not supported on ' . $^O .
```
There is method `sorry` for this. Makes nicer less annoying error messages. ;)

patrickbkr replied:
Done.
